Replicator/Replicator.hh

```diff
@@ -43,7 +43,10 @@ namespace litecore { namespace repl {
         using CloseStatus = blip::Connection::CloseStatus;
         using Options     = litecore::repl::Options;
 
+        static shared_ptr<DBAccess> createDBAccess(C4Database* NONNULL, Options* NONNULL);
```

Comment:
It appears to me little extraneous:  createDBAccess.  vs make_shared<DBAccess>. The latter is more straightforward, saving me a "go to definition" as I encounter it. 
Otherwise, very good.

Reply:
I pulled `createDBAccess` out because we now create `DBAccess`, based on `Options`, in two different places. 

1. In the preexisting `Replicator` constructor, that takes a `C4Database`. It's convenient to keep that because it is used in tests.
2. In `C4RemoteReplicator::createReplicator`.

I wanted to avoid the duplication, but please let me know if I should create `DBAccess` with `make_shared` in both places.

Reply:
Of course, we don't want to remove the old API. I meant to compare make_shared<DBAccess>() with createDBAccess(). They have the same argument signature and users supply same arguments.